crates/router/src/core/mandate.rs

```diff
@@ -410,45 +409,48 @@ where
                         resp.payment_id.clone(),
                         resp.connector.clone(),
                         resp.request.get_setup_mandate_details().cloned(),
-                        maybe_customer,
+                        customer_id,
                         pm_id.get_required_value("payment_method_id")?,
                         mandate_ids,
                         network_txn_id,
-                        get_insensitive_payment_method_data_if_exists(&resp),
+                        get_insensitive_payment_method_data_if_exists(resp),
                         mandate_reference,
                         merchant_connector_id,
                     )? {
                         let connector = new_mandate_data.connector.clone();
                         logger::debug!("{:?}", new_mandate_data);
-                        resp.request
-                        .set_mandate_id(Some(api_models::payments::MandateIds {
-                            mandate_id: Some(new_mandate_data.mandate_id.clone()),
-                            mandate_reference_id: new_mandate_data
-                                .connector_mandate_ids
-                                .clone()
-                            .map(|ids| {
-                                Some(ids)
-                                    .parse_value::<api_models::payments::ConnectorMandateReferenceId>(
-                                        "ConnectorMandateId",
-                                    )
-                                    .change_context(errors::ApiErrorResponse::MandateDeserializationFailed)
-                            })
-                            .transpose()?
-                            .map_or(
-                                new_mandate_data.network_transaction_id.clone().map(|id| {
-                                    api_models::payments::MandateReferenceId::NetworkMandateId(
-                                        id,
-                                    )
-                                }),
-                                |connector_id| Some(api_models::payments::MandateReferenceId::ConnectorMandateId(
-                                    api_models::payments::ConnectorMandateReferenceId {
-                                        connector_mandate_id: connector_id.connector_mandate_id,
-                                        payment_method_id: connector_id.payment_method_id,
-                                        update_history:None,
 
-                                    }
-                                )))
-                        }));
+                        // For GooglePay Mandates
```

Comment:
how are we capturing the `mandate_id` of a newly created mandate?

Reply:
returning the `mandate_id` from the function and updating in the `payment_attempt`.